### PR TITLE
[NodeTypeResolver] Handle optional array shape on native function like with ArrayDimFetch

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector-src/pull/5060

### CODE REVIEW

**Comment by samsonasik** [comment]
Fixed 🎉 /cc @staabm

File: packages/NodeTypeResolver/NodeTypeResolver.php [hunk 1]
```diff
@@ -343,6 +327,62 @@ public function isMethodStaticCallOrClassMethodObjectType(Node $node, ObjectType
...
+         *  if (!empty($parts['host'])) { }
+         */
+        $nativeVariableType = $scope->getNativeType($arrayDimFetch->var);
+        if ($nativeVariableType instanceof MixedType || ($nativeVariableType instanceof ArrayType && $nativeVariableType->getItemType() instanceof MixedType)) {
+            return $originalNativeType;
+        }
+
+        $type = $scope->getType($arrayDimFetch);
+
+        if (! $arrayDimFetch->dim instanceof String_) {
```
Comment (staabm) [inline packages/NodeTypeResolver/NodeTypeResolver.php:358]: This looks fishy. We will surely find array shapes with int keys

Comment (samsonasik) [reply packages/NodeTypeResolver/NodeTypeResolver.php:358]: I think that's need another fixture for next improvement 👍

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I am merging it to have faster feedback to test ;)

**Comment by staabm** [comment]
Thx

**Pull request merged**